File: client/src/main/java/demo/HelloClientApplication.java
package demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import org.springframework.cloud.netflix.feign.EnableFeignClients;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author Spencer Gibb
 */
@SpringBootApplication
@EnableDiscoveryClient
@RestController
@EnableFeignClients
public class HelloClientApplication {
	@Autowired
	HelloClient client;

	@RequestMapping("/")
	public String hello(@RequestParam("name") String name) {
		User user = client.hello1(name);
		User user2 = client.hello2(name);
		System.out.println("从HelloServerh获取到了一个用户" + user + user2);
		return user.toString();
	}

	public static void main(String[] args) {
		SpringApplication.run(HelloClientApplication.class, args);
	}

	@FeignClient("HelloServer")
	interface HelloClient {
		@RequestMapping(value = "/get", method = RequestMethod.GET)
		@ResponseBody
		public User hello1(@RequestParam("name") String name);

		@RequestMapping(value = "/post", method = RequestMethod.POST)
		@ResponseBody
		public User hello2(@RequestParam("name") String name);
	}
}

File: micro-service-hello-client/src/main/java/demo/HelloClientApplication.kt
package demo

import feign.*
import io.seata.rm.datasource.*
import org.springframework.beans.factory.annotation.*
import org.springframework.boot.*
import org.springframework.boot.autoconfigure.*
import org.springframework.boot.autoconfigure.jdbc.*
import org.springframework.cloud.client.discovery.*
import org.springframework.cloud.openfeign.*
import org.springframework.context.annotation.*
import javax.sql.*

@SpringBootApplication(exclude = [DataSourceAutoConfiguration::class])
@EnableDiscoveryClient
@EnableFeignClients
class HelloClientApplication {

  @Bean
  fun feignLoggerLevel(): Logger.Level {
    return Logger.Level.FULL
  }

}

fun main(args: Array<String>) {
  SpringApplication.run(HelloClientApplication::class.java, *args)
}
